peps/pep-0735.rst

```diff
@@ -0,0 +1,476 @@
+  PEP does not define exactly how PEP 723 should be modified, but being
+  consumable by that interface is a stated goal.
+* IDE discovery of requirements data. e.g. VSCode could look for a dependency
+  group named ``test`` to use when running tests.
+
+Regarding Poetry and PDM Dependency Groups
+------------------------------------------
+
+Poetry and PDM already offers a feature which each calls "Dependency Groups",
+but using non-standard data belonging to the ``poetry`` and ``pdm`` tools.
+(PDM also uses extras for some Dependency Groups, and overlaps the notion
+heavily with extras.)
+
+This PEP is not guaranteed to be a perfectly substitutable solution for the
+same problem space for each tool. However, the ideas are extremely similar, and
+it should be possible for Poetry and PDM to support at least some
+PEP-735-standardized Dependency Group configurations using their own Dependency
+Group nomenclature.
+
+A level of interoperability with Poetry and PDM is a goal of this PEP, but
+certain features and behaviors defined here may not be supported by Poetry and
+PDM. Matching the existing Poetry and PDM *semantics* for Dependency Groups is
+a non-goal.
+
+Dependency Groups are not Hidden Extras
+---------------------------------------
+
+One could be forgiven for thinking that Dependency Groups are just extras which
+go unpublished.
+
+However, there are two major features they have which distinguish them from
+extras:
+
+* they support non-package projects
+
+* installation of a dependency group does not imply installation of a package's
+  dependencies (or the package itself)
+
+Specification
+=============
+
+This PEP defines a new section (table) in ``pyproject.toml`` files named
+``requirements``. The ``requirements`` table contains exactly one key,
+``packages``, which is a table. All other keys in ``requirements`` are reserved
+for future use.
+
+The ``packages`` table contains an arbitrary number of user-defined keys, each of
+which has, as its value, a list of requirements specifiers (defined below).
+These keys must match the following regular expression:
+``[a-z0-9][a-z0-9-]*[a-z0-9]``. Meaning that they must be all lower-case
+alphanumerics, with ``-`` allowed only in the middle, and at least
+two characters long. These requirements are chosen so that the normalization
+rules used for pypi package names do not apply, as the names are already
+normalized.
+
+Requirements specifiers will use a definition which extends PEP 508. Therefore,
+it is first necessary to define a string format, a "PEP 735 Dependency". These
+are defined as strings in one of the three following formats:
+
+* a PEP 508 specification, e.g., ``numpy>1``
+* a comma-delimited list of Dependency Group names, enclosed in square
+  brackets, e.g., ``[test,docs]`` to refer to other Dependency Groups named
+  ``test`` and ``docs``
+* a single dot, ``.``, which refers to the current project as a package
+  (similar to ``pip install .``)
```

Comment:
As in you're referring to `project.dependencies`? If so then I would explicitly state that and link to the spec for `project.dependencies` at packaging.python.org .

Reply:
I had meant for it to mean installation of `project.dependencies` + the local package.
My thought was that a dependency group using `.` or `.[foo]` should get the same result as `pip install .` or `pip install '.[foo]'`.

So I think it's slightly broader. Is there a good, more precise term for this?

(And we should handle whether or not the PEP recommends editable installs, but I'm inclined to think it's better to have some expectation about whether or not it's the equivalent of `pip install` or `pip install -e`.)

Reply:
The latest version of the spec replaces this whole idea with an object specification for paths. I think it's a lot more explicit, so I feel comfortable marking this resolved, but obviously the new draft will need review.